experimental/Schemes/src/ToricBlowups/methods.jl

```diff
@@ -1,3 +1,231 @@
+@doc raw"""
+    total_transform(f::ToricBlowupMorphism, I::MPolyIdeal) -> MPolyIdeal
+
+Let $f\colon Y \to X$ be the toric blowup corresponding to a star subdivision along a ray. Let $R$ and $S$ be the Cox rings of $X$ and $Y$, respectively.
+This function returns a homogeneous ideal in $S$ corresponding to the total transform (meaning the scheme-theoretic inverse image) under $f$ of the closed subscheme of $X$ defined by the homogeneous ideal $I$ in $R$.
+
+This is implemented under the following assumptions:
+  * the star subdivision adds a ray (instead of subdividing along an existing ray),
+  * $X$ is an orbifold (meaning its fan is simplicial), and
+  * $X$ has no torus factors (meaning the rays span $N_{\mathbb{R}}$).
+
+# Examples
+```jldoctest
+julia> X = affine_space(NormalToricVariety, 2)
+Normal toric variety
+
+julia> f = blow_up(X, [2, 3])
+Toric blowup morphism
+
+julia> R = cox_ring(X)
+Multivariate polynomial ring in 2 variables over QQ graded by
+  x1 -> 1 by 0 empty matrix
+  x2 -> 1 by 0 empty matrix
+
+julia> (x1, x2) = gens(R)
+2-element Vector{MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}}:
+ x1
+ x2
+
+julia> I = ideal(R, [x1 + x2])
+Ideal generated by
+  x1 + x2
+
+julia> total_transform(f, I)
+Ideal generated by
+  x1*e^2 + x2*e^3
+```
+"""
+function total_transform(f::ToricBlowupMorphism, I::MPolyIdeal)
+  return cox_ring_group_homomorphism(f, I)
+end
+
+@doc raw"""
+    strict_transform(f::ToricBlowupMorphism, I::MPolyIdeal) -> MPolyIdeal
+
+Let $f\colon Y \to X$ be the toric blowup corresponding to a star subdivision along a ray. Let $R$ and $S$ be the Cox rings of $X$ and $Y$, respectively.
+Here "strict transform" means the "scheme-theoretic closure of the complement of the exceptional divisor in the scheme-theoretic inverse image".
+This function returns a homogeneous ideal in $S$ corresponding to the strict transform under $f$ of the closed subscheme of $X$ defined by the homogeneous ideal $I$ in $R$.
+
+This is implemented under the following assumptions:
+  * the star subdivision adds a ray (instead of subdividing along an existing ray),
+  * $X$ is an orbifold (meaning its fan is simplicial), and
+  * $X$ has no torus factors (meaning the rays span $N_{\mathbb{R}}$).
+
+# Examples
+```jldoctest
+julia> X = affine_space(NormalToricVariety, 2)
+Normal toric variety
+
+julia> f = blow_up(X, [2, 3])
+Toric blowup morphism
+
+julia> R = cox_ring(X)
+Multivariate polynomial ring in 2 variables over QQ graded by
+  x1 -> 1 by 0 empty matrix
+  x2 -> 1 by 0 empty matrix
+
+julia> (x1, x2) = gens(R)
+2-element Vector{MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}}:
+ x1
+ x2
+
+julia> I = ideal(R, [x1 + x2])
+Ideal generated by
+  x1 + x2
+
+julia> strict_transform(f, I)
+Ideal generated by
+  x1 + x2*e
+```
+"""
+function strict_transform(f::ToricBlowupMorphism, I::MPolyIdeal)
+  S = cox_ring(domain(f))
+  new_var = S[index_of_new_ray(f)]
+  J = total_transform(f, I)
+  return saturation(J, ideal(new_var))
```

Comment:
```suggestion
  new_var = S[index_of_new_ray(f)]
  J = total_transform(f, I)
  return saturation(J, ideal(S, new_var))
```
Just a nitpick, but to avoid potential future problems.